LeaderboardBackend/Models/ViewModels/LeaderboardViewModel.cs

```diff
@@ -51,11 +54,12 @@ public record LeaderboardViewModel
     /// <summary>
     ///     A collection of `Category` entities for the `Leaderboard`.
     /// </summary>
-    public required IList<CategoryViewModel> Categories { get; init; }
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
```

Comment:
With this change though; we'll have to ensure we do the same thing for any future endpoint with possibly-null relations, for consistency. Imo it's fine to default to an empty array, anyway

Reply:
All this does is not serialize it when there are no cats to be returned.

Reply:
Ergo the field itself won't appear in the return object, right?

Reply:
That could be a problem if frontend is expecting it.

I'd ask the frontend people what they would expect here.
Anecdotally the frontend teams I generally work with would expect an empty array. It just helps avoid safe access checks cluttering up the code, and having to write a new type for leaderboards with and without the field.

Reply:
The frontend code that actually deals with the models is autogenerated so it'll mark the field as potentially absent. If they try to access it without doing a null check, it'll produce a warning.